File: resources/SolrCloud.scala
import io.ino.solrs._
import io.ino.solrs.future.ScalaFutureFactory.Implicit

class SolrCloud extends App {

  // #intro
  val servers = new CloudSolrServers("localhost:2181")
  val solr = AsyncSolrClient.Builder(RoundRobinLB(servers)).build
  // #intro

  // #extended
  val servers = new CloudSolrServers(
    zkHost = "host1:2181,host2:2181",
    zkClientTimeout = 15 seconds,
    zkConnectTimeout = 10 seconds,
    clusterStateUpdateInterval = 1 second,
    defaultCollection = Some("collection1"),
    warmupQueries = WarmupQueries("collection1" => Seq(new SolrQuery("*:*")), count = 10))
  val solr = AsyncSolrClient.Builder(RoundRobinLB(servers)).build
  // #extended

}

File: src/main/paradox/resources/SolrCloud.java
import io.ino.solrs.*;
import java.util.Collections;
import static java.util.concurrent.TimeUnit.*;

class SolrCloud {

  // #intro
  CloudSolrServers<?> servers = CloudSolrServers.builder("localhost:2181").build();
  JavaAsyncSolrClient solr = JavaAsyncSolrClient.builder(new RoundRobinLB(servers)).build();
  // #intro

  // #extended
  CloudSolrServers<?> servers = CloudSolrServers.builder("host1:2181,host2:2181")
    .withZkClientTimeout(15, SECONDS)
    .withZkConnectTimeout(10, SECONDS)
    .withClusterStateUpdateInterval(1, SECONDS)
    .withDefaultCollection("collection1")
    .withWarmupQueries((collection) -> Collections.singletonList(new SolrQuery("*:*")), 10)
    .build();
  JavaAsyncSolrClient solr = JavaAsyncSolrClient.builder(new RoundRobinLB(servers)).build();
  // #extended

}